server/src/main/java/com/google/fhir/gateway/HttpFhirClient.java

```diff
@@ -105,20 +124,45 @@ private void setUri(RequestBuilder builder, String resourcePath) {
   HttpResponse handleRequest(ServletRequestDetails request) throws IOException {
     String httpMethod = request.getServletRequest().getMethod();
     RequestBuilder builder = RequestBuilder.create(httpMethod);
-    setUri(builder, request.getRequestPath());
-    // TODO Check why this does not work Content-Type is application/x-www-form-urlencoded.
-    byte[] requestContent = request.loadRequestContents();
-    if (requestContent != null && requestContent.length > 0) {
-      String contentType = request.getHeader("Content-Type");
-      if (contentType == null) {
-        ExceptionUtil.throwRuntimeExceptionAndLog(
-            logger, "Content-Type header should be set for requests with body.");
+    HttpResponse httpResponse;
+    if (request.getRequestPath().contains(PRACTITIONER_DETAILS)) {
```

Comment:
Can you please take a look at PR #182 and they was custom endpoints are implemented. So the `PRACTITIONER_DETAILS` and `LOCATION_HIERARCHY` endpoints should be implemented outside the server code as this is not a common endpoint that most users need. Instead you can directly implement servlets for handling them.